### DISCUSSION TITLE
Why do Python versions 3.10+ have their minor versions in the package name?

### DISCUSSION THREAD

**Discussion by MajorTanya** [microsoft/winget-pkgs#118361] [Q&A] [answered]
I noticed the other day that Python 3 versions are listed as such:
Output from `winget search Python.Python.3`:
```
Name        ID                 Version  Source
-----------------------------------------------
Python 3    Python.Python.3.9  3.9.13   winget
Python 3    Python.Python.3.8  3.8.10   winget
Python 3    Python.Python.3.7  3.7.9    winget
Python 3    Python.Python.3.6  3.6.8    winget
Python 3    Python.Python.3.5  3.5.4    winget
Python 3    Python.Python.3.4  3.4.4    winget
Python 3    Python.Python.3.3  3.3.5    winget
Python 3    Python.Python.3.2  3.2.5    winget
Python 3.12 Python.Python.3.12 3.12.0b4 winget
Python 3.11 Python.Python.3.11 3.11.4   winget
Python 3.10 Python.Python.3.10 3.10.11  winget
Python 3    Python.Python.3.1  3.1.4    winget
Python 3    Python.Python.3.0  3.0.1    winget
```

You can see how Python 3.10, 3.11, and 3.12 are named in the Name column, whereas those ≤3.9 are just listed as “Python 3”. Was this a design decision for the double-digit minor versions, or is this just happenstance?

**Comment by mdanish-kh** [comment, marked as answer]
The `Python` PackageName is an artifact from an earlier era when Python's minor versions lived under the same PackageId. The old manifests were probably moved as is into their new PackageId without undergoing individual modifications to update their PackageNames.

**Reply by MajorTanya** [reply]
Oh, was that done to avoid showing “available” updates from e.g. Python 3.8 to Python 3.9? I recall that happening a long time ago when I first started using winget. Cool change.

I assume the other versions probably won't be changed for compatibility or something like that, right?

Thanks for your response 👍

**Reply by mdanish-kh** [reply]
> Oh, was that done to avoid showing “available” updates from e.g. Python 3.8 to Python 3.9? I recall that happening a long time ago when I first started using winget. Cool change.

Yes I believe that was the motive. You can read up on the history of the change here:

- https://github.com/microsoft/winget-pkgs/discussions/12004

> I assume the other versions probably won't be changed for compatibility or something like that, right?

Not really, anyone's free to modify those old manifests and correct the PackageNames. Nobody ever pointed it out and it seems like a minor output problem.

**Comment by SpecterShell** [comment]
```
Name            Id                 Version
-------------------------------------------
Python 3.9      Python.Python.3.9  3.9.13
Python 3.8      Python.Python.3.8  3.8.10
Python 3.7      Python.Python.3.7  3.7.9
Python 3.6      Python.Python.3.6  3.6.8
Python 3.5      Python.Python.3.5  3.5.4
Python 3.4      Python.Python.3.4  3.4.4
Python 3.3      Python.Python.3.3  3.3.5
Python 3.2      Python.Python.3.2  3.2.5
Python 3.12     Python.Python.3.12 3.12.1
Python 3.11     Python.Python.3.11 3.11.7
Python 3.10     Python.Python.3.10 3.10.11
Python 3.1      Python.Python.3.1  3.1.4
Python 3.0      Python.Python.3.0  3.0.1
```

All renamed (and enhanced) now.
